graphql-types/src/types-mappings.json

```diff
@@ -0,0 +1,8 @@
+{
```

Comment:
This is the other reason why we use a post-processor for the GraphQL types generation.

We have instances where the GraphQL schema defines a field with an abstract type, but the actual service returns an implementation.

By default, the generated Typescript types will use the abstract type, which won't allow consumers to use the actual data returned by the service.

Here I just implemented a basic example.
The `Geometry` types in the GraphQL schema looks like this:
```graphql
type Geometry {
  type: String!
}
```

but it has only one implementation which is the `Point`:
```graphql
type Geometry {
  type: String!
  coordinates: [Float!]!
}
```

With this configuration we can adjust the generated types to use the implementation type instead of the abstract one.

There are other use cases where an abstract type might have several implementations (`Money` has `HighPrecissionMoney` and `CentPrecissionMoney`). In those cases we are already using just one of the implementations in the test data models, so it wouldn't be much different.

We could even think on extending this first solution so we can decide the swapping per model. Something like this:
```json
  "ctp": [
    {
      "from": "ModelA.fieldA",
      "to": "HightPrecissionMoney"
    },
    {
      "from": "ModelB.fieldA",
      "to": "CentPrecissionMoney"
    }
  ]
}
```

Anyway, let's take it step by step.

Reply:
Seems to work like charm.

Updated here: 9c903f870743350ce9cb61d227fb7e65c53c39b0

Reply:
Do we still need this file if the `useImplementingTypes` option works?